### ISSUE TITLE
2.10.7 and later beta versions: sync gets stuck

### ISSUE THREAD

**Issue by spamies1** [croneter/PlexKodiConnect#1105]
After a few days 2.10.7 stops working, PKC stops syncing and kodi becomes unresponsive, it does not exit and has to be forced closed. 2.10.4 is the last version to work reliably and everything goes back to normal if I downgrade to that version. Debug log attached

[kodi.zip](https://github.com/croneter/PlexKodiConnect/files/4119032/kodi.zip)

**Comment by xinseo** [comment]
Facing the same issue. In my case it tries to sync but Kodi gets stuck during the process. Have to manually exit it.

Nvidia Shield TV

**Comment by croneter** [comment]
I find this bug extremely hard to fix. Any more debug logs might help.

**Comment by spamies1** [comment]
> 
> 
> I find this bug extremely hard to fix. Any more debug logs might help.

You want more logs from me or others with the same problem?

**Comment by croneter** [comment]
@spamies1 Anyone, really, having this issue 😉

**Comment by spamies1** [comment]
I installed 2.10.8 and did a DB reset. I'll post logs when/if things go bad.

**Comment by croneter** [comment]
Still happening with 2.10.8? I have troubles reproducing this issue...

**Comment by xinseo** [comment]
> Still happening with 2.10.8? I have troubles reproducing this issue...

Resetting database fixed it for me on 2.10.7

**Comment by croneter** [comment]
I have a feeling that this bug could be a race condition. Which will make it very hard to reproduce and might lead to days of using Plex without any problems...

Please report back with DEBUG logs if you encounter this bug again 😉

**Comment by xinseo** [comment]
> I have a feeling that this bug could be a race condition. Which will make it very hard to reproduce and might lead to days of using Plex without any problems...
> 
> Please report back with DEBUG logs if you encounter this bug again

in my case, Kodi would get stuck every time i turned it on. Resetting databse definitely fixed it.

**Comment by croneter** [comment]
Did anyone of you encounter this issue again with 2.10.9?

**Comment by spamies1** [comment]
seems to be working better for me. thanks

**Comment by Asgardsurfer** [comment]
I also got the freeze this morning, Kodi was completely unresponsive, I had to unplug and plug back in to get it to work again.

**Comment by croneter** [comment]
@Asgardsurfer  If **KODI** got completely unresponsive, this ain't PKC's fault 😆. Theoretically, PKC should not be able to put Kodi in an unresponsive state. 

In other words: your crash of Kodi was likely unrelated to this issue

**Comment by croneter** [comment]
@Asgardsurfer Please switch to PKC beta and version 2.10.9 (latest beta version)

**Comment by Asgardsurfer** [comment]
Thanks croneter for the explanation!
Will update to the latest beta version, thank you again and thank you for your hard work and dedication!

**Comment by Cyriltra** [comment]
Hello, i believe i have the same issue (in 2.10.9), using Coreelec. (I posted a message on plex.tv forum, pseudo PLOUF).
The problem is as soon as i enter some menu (library, app browser etc.) the screen freeze and I have to restart kodi, but the issue keeps coming back.

**Comment by Cyriltra** [comment]
I managed to enable debug logs via command line and restart kodi.
right after the restart and before I enter ANY menu in kodi, the log i have te below that appears every sec:
```
2020-02-08 14:40:52.257 T:3371160384   DEBUG: PLEX.plexgdm: Recieved UDP packet from [('192.168.50.10', 43717)] containing [M-SEARCH * HTTP/1.1]
2020-02-08 14:40:52.258 T:3371160384   DEBUG: PLEX.plexgdm: Detected client discovery request from ('192.168.50.10', 43717).  Replying
2020-02-08 14:40:52.259 T:3371160384   DEBUG: PLEX.plexgdm: Sending registration data HTTP/1.0 200 OK
2020-02-08 14:40:57.262 T:3371160384   DEBUG: PLEX.plexgdm: Recieved UDP packet from [('192.168.50.10', 43717)] containing [M-SEARCH * HTTP/1.1]
2020-02-08 14:40:57.263 T:3371160384   DEBUG: PLEX.plexgdm: Detected client discovery request from ('192.168.50.10', 43717).  Replying
2020-02-08 14:40:57.264 T:3371160384   DEBUG: PLEX.plexgdm: Sending registration data HTTP/1.0 200 OK
2020-02-08 14:41:02.267 T:3371160384   DEBUG: PLEX.plexgdm: Recieved UDP packet from [('192.168.50.10', 43717)] containing [M-SEARCH * HTTP/1.1]
2020-02-08 14:41:02.268 T:3371160384   DEBUG: PLEX.plexgdm: Detected client discovery request from ('192.168.50.10', 43717).  Replying
2020-02-08 14:41:02.269 T:3371160384   DEBUG: PLEX.plexgdm: Sending registration data HTTP/1.0 200 OK
2020-02-08 14:41:07.273 T:3371160384   DEBUG: PLEX.plexgdm: Recieved UDP packet from [('192.168.50.10', 43717)] containing [M-SEARCH * HTTP/1.1]
2020-02-08 14:41:07.274 T:3371160384   DEBUG: PLEX.plexgdm: Detected client discovery request from ('192.168.50.10', 43717).  Replying
2020-02-08 14:41:07.275 T:3371160384   DEBUG: PLEX.plexgdm: Sending registration data HTTP/1.0 200 OK
```

i then navigate in some menu ( I try to go to PKC settings to disable it), but I cant reach it in time before the screen freeze. Once the screen is frozen, the same logs appear every few sec.
I try to go back (using my remote or keyboard), i can see in the log that kodi receive the request for it, but nothing change on screen, still appear as frozen

```
2020-02-08 14:45:27.652 T:3371160384   DEBUG: PLEX.plexgdm: Recieved UDP packet from [('192.168.50.10', 43717)] containing [M-SEARCH * HTTP/1.1]
2020-02-08 14:45:27.653 T:3371160384   DEBUG: PLEX.plexgdm: Detected client discovery request from ('192.168.50.10', 43717).  Replying
2020-02-08 14:45:27.654 T:3371160384   DEBUG: PLEX.plexgdm: Sending registration data HTTP/1.0 200 OK
2020-02-08 14:45:32.655 T:3371160384   DEBUG: PLEX.plexgdm: Recieved UDP packet from [('192.168.50.10', 43717)] containing [M-SEARCH * HTTP/1.1]
2020-02-08 14:45:32.656 T:3371160384   DEBUG: PLEX.plexgdm: Detected client discovery request from ('192.168.50.10', 43717).  Replying
2020-02-08 14:45:32.657 T:3371160384   DEBUG: PLEX.plexgdm: Sending registration data HTTP/1.0 200 OK
2020-02-08 14:45:37.660 T:3371160384   DEBUG: PLEX.plexgdm: Recieved UDP packet from [('192.168.50.10', 43717)] containing [M-SEARCH * HTTP/1.1]
2020-02-08 14:45:37.661 T:3371160384   DEBUG: PLEX.plexgdm: Detected client discovery request from ('192.168.50.10', 43717).  Replying
2020-02-08 14:45:37.662 T:3371160384   DEBUG: PLEX.plexgdm: Sending registration data HTTP/1.0 200 OK
2020-02-08 14:45:39.124 T:4080005952   DEBUG: CLibInputKeyboard::ProcessKey - using delay: 500ms repeat: 33ms
2020-02-08 14:45:39.125 T:3362767680   DEBUG: Thread Timer start, auto delete: false
2020-02-08 14:45:39.258 T:3362767680   DEBUG: Thread Timer 3362767680 terminating
2020-02-08 14:45:39.649 T:4080005952   DEBUG: CLibInputKeyboard::ProcessKey - using delay: 500ms repeat: 33ms
2020-02-08 14:45:39.649 T:3362767680   DEBUG: Thread Timer start, auto delete: false
2020-02-08 14:45:39.817 T:3362767680   DEBUG: Thread Timer 3362767680 terminating
2020-02-08 14:45:39.925 T:4080005952   DEBUG: CLibInputKeyboard::ProcessKey - using delay: 500ms repeat: 33ms
2020-02-08 14:45:39.926 T:3362767680   DEBUG: Thread Timer start, auto delete: false
2020-02-08 14:45:40.033 T:3362767680   DEBUG: Thread Timer 3362767680 terminating
2020-02-08 14:45:40.143 T:4080005952   DEBUG: CLibInputKeyboard::ProcessKey - using delay: 500ms repeat: 33ms
2020-02-08 14:45:40.143 T:3362767680   DEBUG: Thread Timer start, auto delete: false
2020-02-08 14:45:40.253 T:3362767680   DEBUG: Thread Timer 3362767680 terminating
2020-02-08 14:45:40.364 T:4080005952   DEBUG: CLibInputKeyboard::ProcessKey - using delay: 500ms repeat: 33ms
2020-02-08 14:45:40.364 T:3362767680   DEBUG: Thread Timer start, auto delete: false
2020-02-08 14:45:40.474 T:3362767680   DEBUG: Thread Timer 3362767680 terminating
2020-02-08 14:45:40.582 T:4080005952   DEBUG: CLibInputKeyboard::ProcessKey - using delay: 500ms repeat: 33ms
2020-02-08 14:45:40.582 T:3362767680   DEBUG: Thread Timer start, auto delete: false
2020-02-08 14:45:40.692 T:3362767680   DEBUG: Thread Timer 3362767680 terminating
2020-02-08 14:45:40.800 T:4080005952   DEBUG: CLibInputKeyboard::ProcessKey - using delay: 500ms repeat: 33ms
2020-02-08 14:45:40.801 T:3362767680   DEBUG: Thread Timer start, auto delete: false
2020-02-08 14:45:40.883 T:3362767680   DEBUG: Thread Timer 3362767680 terminating
2020-02-08 14:45:42.666 T:3371160384   DEBUG: PLEX.plexgdm: Recieved UDP packet from [('192.168.50.10', 43717)] containing [M-SEARCH * HTTP/1.1]
2020-02-08 14:45:42.667 T:3371160384   DEBUG: PLEX.plexgdm: Detected client discovery request from ('192.168.50.10', 43717).  Replying
2020-02-08 14:45:42.667 T:3371160384   DEBUG: PLEX.plexgdm: Sending registration data HTTP/1.0 200 OK
```

>

**Comment by croneter** [comment]
@Cyriltra your issue is unrelated and your log snippets are useless, unfortunately - I need the entire DEBUG log.

Please try using LibreElec as I had issues with CoreElec in the past. If your issue persists, open a new issue here with the entire DEBUG log.

**Comment by croneter** [comment]
@Cyriltra judging from the Plex forum screenshots, you got the same issue. Kodi shouldNOT become unresponsive though. Thanks for sharing your story log :wink:

**Comment by Cyriltra** [comment]
sure no pb, let me know if you need me to run some tests (if I can as i'm stuck)
If you know by the way how to disable/enable PBK via SSH command line, let me know.
if I can do it, i can check first if the issue is really linked to PKC or not.
If it's PKC, i could try to reset all db to see if it fix the issue, at least temporarily.

FYI, i think my last db reset was at least several weeks ago and i run kodi/CE on a tvbox, hence always on.

Thank you

**Comment by croneter** [comment]
@Cyriltra  Restarting Kodi does not fix your issue? It did fix mine and now I can't reproduce anymore :see_no_evil:

**Comment by Cyriltra** [comment]
Hardware reset have fixed the issue for me too, at least temporarily.
WHat i did before was to restart kodi via SSH (without rebooting), this did not fix the issue.

**Comment by spamies1** [comment]
I've started having the same problem, kodi locks up and has to be forced closed. This is on an nvidia shield.  I'm not able to do a PKC database reset. Debug logs attached. 

[kodi.log](https://github.com/croneter/PlexKodiConnect/files/4174718/kodi.log)

**Comment by croneter** [comment]
BETA version 2.10.10 should help, could you please test it?

**Comment by spamies1** [comment]
> 
> 
> BETA version 2.10.10 should help, could you please test it?

Installed and did a database reset. I'll post debug logs if things go bad.

**Comment by croneter** [comment]
Database reset should not be necessary

**Comment by spamies1** [comment]
Looks like its stopped syncing items. Added new movies to plex and they don't show up in kodi. Tried a manual PKC library sync but no change, debug logs attached. At least kodi doesn't become unresponsive.

[kodi.log](https://github.com/croneter/PlexKodiConnect/files/4210612/kodi.log)

**Comment by croneter** [comment]
OK, I think I finally found the bugger. Please test 2.10.11 and let me know. Thanks so much for your patience!

**Comment by spamies1** [comment]
> OK, I think I finally found the bugger. Please test 2.10.11 and let me know. Thanks so much for your patience!

Installed. Thanks for your great software, can't imagine using the plex client instead of PKC and kodi.

**Comment by croneter** [comment]
Any issues yet..? How about you guys, @xinseo, @Asgardsurfer , @Cyriltra   ?

**Comment by xinseo** [comment]
> Any issues yet..? How about you guys, @xinseo, @Asgardsurfer , @Cyriltra ?

My problem was fixed when I resetted databse. Haven't had any issues since.

**Comment by Asgardsurfer** [comment]
Hey croneter, using your latest update 2.10.11, i’ve had it running for two days now and everything seems just right. If I experience any hiccups, I will be sure to let you know. Great work!

**Comment by Cyriltra** [comment]
Hi @croneter,

i had the sync lock issue happened again with 2.10.10 => I came back from work and noticed kodi/pkc had the "sync" banner stuck on top of the screen.
I then upgraded to 2.10.11 then reboot and I still have it.

I DM you my log file

Thank you

**Comment by croneter** [comment]
@Cyriltra seems like your upgrade process was botched. Pkc should still sync fine now. Please let me know if the sync gets stuck again

**Comment by croneter** [comment]
This should be fixed with 2.10.12. Let me know here if this is not the case.
